### ISSUE TITLE
immutable.js updating/setting complicated Map/List

### ISSUE THREAD

**Issue by kocur4d** [facebook/immutable-js#756]
I have a problem I am trying to solve and I have a question about immutable `updateIn` function.

Lets say I have a Map:

```javascript
var testing = Immutable.fromJS({
  state: {
    name: 'secret',
    elements: [
      {
        id: 678,
        name: 'first'
      },
      {
        id: 689,
        name: 'first'
      },
      {
        id: 699,
        name: 'first'
      }
    ]
  }
})
```

Using `updateIn` to update the name of the one of the elements I can:

```javascript
testing.setIn(
  ['state', 'elements', 1, 'name'], 
  'new element index 1 name'
)
```

The only think I have a problem with - I have found workarounds so its more convenience - is that I need to know index of the object I want to update and sometimes I don't have it at the moment when i create the array but I always have the id of this element.  

Do you think i could be beneficial to allow a function inside the `keyPath` of the `List.updateIn` function:

```javascript
testing.setIn(
  ['state', 'elements', (object ) = > object.id == 689, 'name'], 
  'new element index 1 name'
)
```

Basically Immutable.List item could check if the type of argument of the `keyPath` element is a function and if it is inject the current object into it and return `id` of that element so the `keyPath` end up being like this:

```javascript
['state', 'elements', 1, 'name']
```

In this case I can update more complicated object using the same way:

```javascript
['ads', (object ) = > object.id == 7 , 'creative', 'object_stor_spec', 'link_data', 'child_attachments', (object ) = > object.id == 22, 'name']
```

I didn't think about implementation yet - but i think it is possible, so the functions code examples are just a drafts not the final functions you ill be passing down.

**Comment by kocur4d** [comment]
closing as probably out of the context of immutable

**Comment by MarkusPint** [comment]
@kocur4d I have exactly the same problem. Did you find any good solution?

**Comment by brunoreis** [comment]
+1

**Comment by leiming** [comment]
+1

**Comment by Lohann** [comment]
+1

**Comment by frontenddeveloping** [comment]
+1

**Comment by jebarjonet** [comment]
+1

**Comment by antsav** [comment]
+1

**Comment by jmancherje** [comment]
+1

**Comment by frodrigueza** [comment]
+1

**Comment by reduxdj** [comment]
+1

**Comment by TwinLight** [comment]
+ 1, Sorry is there any solution to this

**Comment by leebyron** [comment]
Locking this closed conversation.

For future reference, if the intent is to find an item with a particular id, I suggest a `Map` keyed by id instead of a `List` - lookups will be much cheaper than walking the List to look for the id.
